### PR TITLE
[IMP] Detect removed tags in update data

### PR DESCRIPTION
**Pull request by StefanRijnhart**
Detect problems like #237

### CODE REVIEW

File: scripts/compare_noupdate_xml_records.py [hunk 1]
```diff
@@ -185,20 +196,29 @@ def main(argv=None):
...
+            for key in record_new_dict.keys():
+                if not record_old.xpath(key):
+                    element.append(deepcopy(record_new_dict[key]))
 
         if len(element):
             data.append(element)
 
+    data.append(etree.Comment("Changes in noupdate data"))
+    for xml_id, record_new in new_noupdate.items():
+        compare_record(xml_id, record_new)
```
Comment (pedrobaeza) [inline scripts/compare_noupdate_xml_records.py:216]: I don't get it. Here you are only reassigning old_record, but not doing anything with it.

Comment (StefanRijnhart) [inline scripts/compare_noupdate_xml_records.py:216]: I'm not assigning old_record, but new_record and I'm passing this to an inline method that adds to the list of elements which is scoped under the main method.

Comment (pedrobaeza) [inline scripts/compare_noupdate_xml_records.py:216]: But in `compare_record` method record_new is not handled...

Comment (StefanRijnhart) [inline scripts/compare_noupdate_xml_records.py:216]: Are we talking about the same compare_record? In the inline function that I am talking about, record_new is a function argument which is referred to several times.

Comment (pedrobaeza) [inline scripts/compare_noupdate_xml_records.py:216]: Sorry, I was confused about the diff and the indentation, which is very difficult to follow. I have seen full code and now it's clear.

**Comment by pedrobaeza** [comment]
:+1:

**Comment by StefanRijnhart** [comment]
I can imagine! I made minimal changes for this fix but I should probably refactor it next. Thanks for the review.

**Comment by hbrunn** [comment]
:+1:

**Pull request merged**